docs/api-reference/next/router.md

```diff
@@ -274,53 +317,82 @@ You can listen to different events happening inside the Router. Here's a list of
 - `hashChangeStart(url)` - Fires when the hash will change but not the page
 - `hashChangeComplete(url)` - Fires when the hash has changed but not the page
 
-> Here `url` is the URL shown in the browser. If you call `Router.push(url, as)` (or similar), then the value of `url` will be `as`.
+> Here `url` is the URL shown in the browser. If you call `router.push(url, as)` (or similar), then the value of `url` will be `as`.
+
+#### Usage
 
-For example, to listen to the router event `routeChangeStart`, do the following:
+For example, to listen to the router event `routeChangeStart`, Open or create `pages/_app.js` and subscribe to the event, like so:
 
 ```jsx
-import Router from 'next/router'
+import { useEffect } from 'react'
+import { useRouter } from 'next/router'
 
-const handleRouteChange = (url) => {
-  console.log('App is changing to: ', url)
-}
+export default function MyApp({ Component, pageProps }) {
+  const router = useRouter()
 
-Router.events.on('routeChangeStart', handleRouteChange)
-```
+  useEffect(() => {
+    const handleRouteChange = (url) => {
+      console.log('App is changing to: ', url)
+    }
 
-If you no longer want to listen to the event, unsubscribe with the `off` method:
+    router.events.on('routeChangeStart', handleRouteChange)
 
-```jsx
-import Router from 'next/router'
+    // If the component is unmounted, unsubscribe
+    // from the event with the `off` method:
+    return () => {
+      router.events.off('routeChangeStart', handleRouteChange)
+    }
+  }, [])
 
-Router.events.off('routeChangeStart', handleRouteChange)
+  return <Component {...pageProps} />
+}
 ```
 
+> We use a [Custom App](/docs/advanced-features/custom-app) (`pages/_app.js`) for this example to subscribe to the event because it's not unmounted on page navigations, but you can subscribe to router events on any component in your application.
```

Comment:
```suggestion
> We use a [Custom App](/docs/advanced-features/custom-app.md) (`pages/_app.js`) for this example to subscribe to the event because it's not unmounted on page navigations, but you can subscribe to router events on any component in your application.
```